public/resources/remix/index.js

```diff
@@ -6,122 +6,136 @@
-      cache: false,
-      accepts: "text/html",
-      dataType: "html",
-      data: {
-        host: metadata.host,
-        id: metadata.projectId,
-        title: metadata.projectTitle,
-        author: metadata.projectAuthor,
-        updated: metadata.dateUpdated
+  function injectDetailsBar(metadata, callback) {
+    var xmlhttp = new XMLHttpRequest();
+    xmlhttp.responseType = "text";
+    var url = metadata.host + "/projects/remix-bar";
+    var data = {
+      host: metadata.host,
+      id: metadata.projectId,
+      title: metadata.projectTitle,
+      author: metadata.projectAuthor,
+      updated: metadata.dateUpdated
+    };
+
+    xmlhttp.onreadystatechange = function() {
+      if (xmlhttp.readyState == XMLHttpRequest.DONE ) {
+        var response = xmlhttp.response;
+        var textStatus = xmlhttp.statusText;
+        if (xmlhttp.status == 200) {
+          document.body.insertBefore(response, document.body.firstChild);
+	   callback(null);
+        } else if (xmlhttp.status == 400) {
+	   callback(textStatus);
+        }
       }
-    })
-    .done(function(response) {
-      $(response).prependTo("body");
-      callback(null, $);
-    })
-    .fail(function(jqXHR, textStatus) {
-      callback(textStatus);
-    });
+    };
+
+    xmlhttp.open("POST", url, true);
+    xmlhttp.send(data);
   }
 
-  function injectStyleSheets($, metadata) {
-    var stylesheets =
-      "<link href=\"" + metadata.host + "/resources/remix/clean-slate.min.css\" rel=\"stylesheet\">\n" +
-      "<link href=\"" + metadata.host + "/resources/remix/style.css\" rel=\"stylesheet\">\n";
-    $("head").append(stylesheets);
+  function injectStyleSheets(metadata) {
+    var stylesheet = document.createElement("link");
+    stylesheet.rel = "stylesheet";
+    stylesheet.href = metadata.host + "/resources/remix/clean-slate.min.css";
+    document.head.appendChild(stylesheet);
+
+    stylesheet = document.createElement("link");
+    stylesheet.rel = "stylesheet";
+    stylesheet.href = metadata.host + "/resources/remix/style.css";
+    document.head.appendChild(stylesheet);
+  }
+
+  function grep(items, callback) {
+    var filtered = [], len = items.length, i = 0;
+    for (i; i < len; i++) {
+      var item = items[i];
+      var cond = callback(item);
+      if (cond) {
+        filtered.push(item);
+      }
+    }
+    return filtered;
   }
 
-  function getMetadata($) {
+  function getMetadata() {
     var metadata = {};
-    var metaTags = $("meta");
+    var metaTags = document.getElementsByTagName("meta");
 
-    $.grep(metaTags, function(elem, index) {
-      return /^data-remix-.+/.test($(metaTags[index]).attr("name"));
+    grep(metaTags, function(metaTag) {
+      return /^data-remix-.+/.test((metaTag).getAttribute("name"));
     })
     .forEach(function(metaTag) {
-      metaTag = $(metaTag);
-      var key = metaTag.attr("name").replace(/^data-remix-/, "");
+      var key = metaTag.getAttribute("name").replace(/^data-remix-/, "");
```

Comment:
if we use the preceding code, we are already guaranteed that any element that makes it through the filter has a value for `name` that starts with `data-remix-` so we don't need a regular expression here anymore either, we can simply say `var key = metaTag.getAttribute("name").replace("data-remix-", "");`

That said, the argument to this function should probably be metaElement, since we're passing around elements, not tags (tags are strings)